### ISSUE TITLE
After recalling a previous minimum, validation is performed immediately again.

### ISSUE THREAD

**Issue by senarvi** [senarvi/theanolm#9]
After performance on validation set has ceased to improve and the previously best state is restored, training is at a validation point. Immediately after restoring the state, validation is performed again:

    [273669] 10 samples collected, statistic 921.53.
    [273669] Validation set cost history: 930.5 958.2 953.5 934.9 938.2 927.2 927.1 926.4 [916.1] 923.1 919.7 918.2 918.6 916.9 918.7 929.7 926.2 929.1 924.1 921.5
    reset_state
    [191563] Validation set cost history: 949.0 939.3 937.3 942.5 935.0 937.1 939.4 948.2 942.8 931.4 929.9 930.5 958.2 953.5 934.9 938.2 927.2 927.1 926.4 [916.1]
    decrease_learning_rate
    [191568] 5 samples collected, statistic 916.10.
    [191568] Validation set cost history: 939.3 937.3 942.5 935.0 937.1 939.4 948.2 942.8 931.4 929.9 930.5 958.2 953.5 934.9 938.2 927.2 927.1 926.4 [916.1] 916.1
